joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -40,6 +38,25 @@ trait ProgramSummary[T <: TypeLike[_, _]] {
 
 }
 
+object ProgramSummary {
+
+  /** Combines two namespace-to-type maps.
+    */
+  def combine[T <: TypeLike[_, _]](a: Map[String, Set[T]], b: Map[String, Set[T]]): Map[String, Set[T]] = {
+    val accumulator = mutable.HashMap.from(a)
+    val allKeys     = accumulator.keySet ++ b.keySet
+
+    allKeys.foreach(k =>
```

Comment:
Nit: since `accumulator` is already taken from `a`, we could probably just traverse `b`'s keys, right?

Reply:
Good point, will implement the suggestion!